### PR TITLE
Turn splits inside out

### PR DESCRIPTION
**Pull request by tylerwilliams**

### CODE REVIEW

File: enterprise/server/raft/replica/replica.go [hunk 1]
```diff
@@ -743,194 +669,71 @@ func (sm *Replica) deleteStoredFiles(start, end []byte) error {
...
-		LowerBound: keys.Key(start),
-		UpperBound: keys.Key(end),
+func (sm *Replica) copyStoredFiles(req *rfpb.CopyStoredFilesRequest) (*rfpb.CopyStoredFilesResponse, error) {
+	appliesToThisReplica := sm.splitAppliesToThisReplica(req.GetSourceRange())
+	if !appliesToThisReplica {
+		sm.log.Debugf("Got copyStoredFiles cmd but it doesn't apply to this replica; skipping")
+		return &rfpb.CopyStoredFilesResponse{}, nil
+	}
+
+	localReplica, err := sm.store.GetReplica(req.GetTargetRangeId())
```
Comment (vadimberezniker) [inline enterprise/server/raft/replica/replica.go:679]: can we call this targetReplica or destReplica? both of the replicas involved are local

File: enterprise/server/raft/replica/replica.go [hunk 2]
```diff
@@ -1695,6 +1513,52 @@ func (sm *Replica) ApplySnapshotFromReader(r io.Reader, db ReplicaWriter) error
...
+	readBuf := bufio.NewReader(r)
+	go func() {
+		defer close(ch)
+		for {
+			r, count, err := readDataFromReader(readBuf)
+			if err != nil {
+				if err == io.EOF {
+					break
+				}
+				ch <- Record{Error: err}
```
Comment (vadimberezniker) [inline enterprise/server/raft/replica/replica.go:1533]: looks like you are missing `continue` here and for the other error cases in this function

File: proto/raft.proto [hunk 3]
```diff
@@ -184,6 +184,17 @@ message SplitReleaseRequest {
 }
 message SplitReleaseResponse {}
 
+// A request to hold or renew the split lock, blocking all writes. The split
```
Comment (vadimberezniker) [inline proto/raft.proto:187]: fix comment

Comment (tylerwilliams) [reply proto/raft.proto]: Done.

File: enterprise/server/raft/replica/replica.go (hunk 2 above)
Comment (tylerwilliams) [reply enterprise/server/raft/replica/replica.go:1537]: Added break -- any error in this function stops the show

File: enterprise/server/raft/replica/replica.go (hunk 1 above)
Comment (tylerwilliams) [reply enterprise/server/raft/replica/replica.go]: Done

**Pull request merged**